packages/cli-core/src/package-manager-controller/package_manager_controller_factory.test.ts

```diff
@@ -62,8 +62,24 @@ void describe('packageManagerControllerFactory', () => {
 
       assert.throws(
         () => packageManagerControllerFactory.getPackageManagerController(),
-        Error,
-        'Package Manager unsupported is not supported.'
+        {
+          message: 'Package Manager unsupported is not supported.',
+        }
+      );
+    });
+
+    void it('should throw an error for pnpm on Windows', () => {
+      const userAgent = 'pnpm/1.0.0 node/v15.0.0 darwin x64';
+      process.env.npm_config_user_agent = userAgent;
+      const packageManagerControllerFactory =
+        new PackageManagerControllerFactory(packageRoot, printer, 'win32');
+
+      assert.throws(
+        () => packageManagerControllerFactory.getPackageManagerController(),
+        {
+          message:
+            "PNPM can't create amplify on Windows. Please switch to NPM or Yarn. \nDetails: https://pnpm.io/faq#but-the-nested-node_modules-approach-is-incompatible-with-windows",
```

Comment:
```suggestion
            "Amplify does not support PNPM on Windows.\nDetails: https://pnpm.io/faq#but-the-nested-node_modules-approach-is-incompatible-with-windows",
```

I think we can remove the "switch" suggestion (we could as well suggest to switch OS :-) ). 
As for the link, I think it would be better to link to our docs page that explains the problem rather than external page (we don't control external page - it can disappear).